x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/LabelFieldProducer.java

```diff
@@ -144,4 +147,24 @@ static class AggregateMetricFieldProducer extends LabelLastValueFieldProducer {
             super(name, new LastValueLabel(metric.name()));
         }
     }
+
+    public static class HistogramLastLabelFieldProducer extends LabelLastValueFieldProducer {
+        HistogramLastLabelFieldProducer(String name) {
+            super(name);
+        }
+
+        @Override
+        public void write(XContentBuilder builder) throws IOException {
+            if (isEmpty() == false) {
+                final HistogramValue histogramValue = (HistogramValue) label.get();
+                final List<Double> values = new ArrayList<>();
+                final List<Integer> counts = new ArrayList<>();
+                while (histogramValue.next()) {
+                    values.add(histogramValue.value());
+                    counts.add(histogramValue.count());
+                }
+                builder.startObject(name()).field("counts", counts.toArray()).field("values", values.toArray()).endObject();
```

Comment:
I agree. In the current case, we can create only the array of longs and serialize doubles as we iterate. 
However, I don't think this would yield any real performance gains. So, I think this code is fine.